smartmeter_datacollector/smartmeter/siemens_td3511.py

```diff
@@ -0,0 +1,255 @@
+#
+# Copyright (C) 2024 IBW Technik AG
+# This file is part of smartmeter-datacollector.
+#
+# SPDX-License-Identifier: GPL-2.0-only
+# See LICENSES/README.md for more information.
+#
+import asyncio
+import logging
+import re
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from typing import Callable, List, Optional
+
+import aioserial
+import serial
+
+from .meter import Meter, MeterError
+from .meter_data import MeterDataPoint, MeterDataPointType, MeterDataPointTypes
+from .reader import Reader, ReaderError
+from .serial_reader import SerialConfig
+
+LOGGER = logging.getLogger("smartmeter")
+
+
+class SiemensTD3511(Meter):
+    BAUDRATE = 19200
+
+    def __init__(self, port: str,
+                 baudrate: int = BAUDRATE,
+                 use_system_time: bool = False) -> None:
+        super().__init__()
+        serial_config = SerialConfig(
+            port=port,
+            baudrate=baudrate,
+            data_bits=serial.SEVENBITS,
+            parity=serial.PARITY_EVEN,
+            stop_bits=serial.STOPBITS_ONE,
+            termination=b"\r\n"
+        )
+        try:
+            self._parser = SiemensParser(use_system_time)
+            self._serial = SiemensSerialReader(serial_config, self._data_received)
+        except ReaderError as ex:
+            LOGGER.fatal("Unable to setup serial reader for Siemens TD3511. '%s'", ex)
+            raise MeterError("Failed setting up Siemens TD3511.") from ex
+
+        LOGGER.info("Successfully set up Siemens TD3511 smart meter on '%s'.", port)
+
+    async def start(self) -> None:
+        await self._serial.start_and_listen()
+
+    def _data_received(self, received_data: bytes) -> None:
+        if not received_data:
+            return
+        if received_data != self._serial.TERMINATION_FLAG:
+            self._parser.append_to_buffer(received_data)
+            return
+
+        data_points = self._parser.parse_data_objects(self._serial.timestamp)
```

Comment:
`self._serial.timestamp` might be `None`. Check for `None` or consider my other comment about timestamps: https://github.com/scs/smartmeter-datacollector/pull/69/files#r1750096111

Reply:
Ok, it's easier to set the timestamp as fallback  in parse_data_objects() so not having to set it in the Reader. I adopted your former proposal.